### ISSUE TITLE
Support for default workload_profile

### ISSUE THREAD

**Issue by michalkrzych** [hashicorp/terraform-provider-azurerm#24198]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

Thank you for [recently adding support for `workload_profile` in 3.84](https://github.com/hashicorp/terraform-provider-azurerm/pull/23478)

I notice this creates a dedicated profile which needs to be deleted after deployment. While it's not a massive issue I can't see why terraform couldn't support what az cli already does with [--enable-workload-profiles](https://learn.microsoft.com/en-us/cli/azure/containerapp/env?view=azure-cli-latest#az-containerapp-env-create-examples).

```
az containerapp env create -n MyContainerappEnvironment -g MyResourceGroup \
    --location eastus2 --enable-workload-profiles
```

This enables workload profiles and adds a default [Consumption](https://learn.microsoft.com/en-us/azure/container-apps/environment#types) (not to be confused with Consumption Only) so no deletion is required post deployment.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/6545828/a3c3d522-a3f9-44d8-b50b-cd426da16fff)

It would be great if we could provide an additional parameter to [container_app_environment](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/container_app_environment) sort of like this:

```
resource "azurerm_container_app_environment" "example" {
  name = "my-environment"
  location = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id
  **enable-workload-profiles = true**
}
```

### New or Affected Resource(s)/Data Source(s)

azurerm_container_app_environment

### Potential Terraform Configuration

```hcl
resource "azurerm_container_app_environment" "example" {
  name = "my-environment"
  location = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  log_analytics_workspace_id = azurerm_log_analytics_workspace.example.id
  enable-workload-profiles = true
}
```

### References

https://github.com/hashicorp/terraform-provider-azurerm/pull/23478

**Comment by kaganmersin** [comment]
@rcskosir Is there any update for this issue ? I believe this issue is really crucial since Consumption profile type is one of the fundamental feature of the container apps. I see one PR is still open related this issue.  
https://github.com/hashicorp/terraform-provider-azurerm/pull/24277

**Comment by jackofallops** [comment]
> @rcskosir Is there any update for this issue ? I believe this issue is really crucial since Consumption profile type is one of the fundamental feature of the container apps. I see one PR is still open related this issue. #24277

Hi @kaganmersin et al - The `Consumption` profile is always present, there can be only one, and it cannot be removed or managed, so it is not managed by the provider at this time (if it becomes removable/changeable, we'll update accordingly). The docs could be clearer on this. Adding to the confusion is that the portal will not display the workload profile option in the blade and display `Consumption Only`, so it appears to not be there. Adding a workload profile to your Terraform configuration for an existing environment will successfully add this to the resource, and then the portal will also display the option thereafter.

The short version of this is that workload profiles are always enabled behind the scenes, and the cli / portal options are, imo, a little misleading.

Does this help / cover your use case?

Thanks!
